python/ql/lib/design.md

```diff
@@ -15,10 +15,10 @@ The analysis is built up in layers. the stack looks like this:
 
 ## Avoiding non-monotonic recursion
 
-Given the many interactivg layers, it is imprtant to decie which predicates are allowed to be mutually recursive in order to avoid non-monotonic recursion when negation is used to express the predicates.
-As an example, we have defined local source as those whcih do not receive local flow. This means that the local flow relation is not allowed to be recursive with anything depending on local sources.
+Given the many interactive layers, it is important to decie which predicates are allowed to be mutually recursive in order to avoid non-monotonic recursion when negation is used to express the predicates.
```

Comment:
```suggestion
Given the many interacting layers, it is important to decide which predicates are allowed to be mutually recursive in order to avoid non-monotonic recursion when negation is used to express the predicates.
```
Indeed, this should be `interacting`. Also, the typo `decie` somehow slipped through as well. 🙂

Reply:
Well, there are >700 tokens for this slice of the repository, and I'm just letting Google Sheets do my spell checking (especially given the slices) -- the only manual stuff I'm doing is addressing [additional alerts that check-spelling generates](https://github.com/check-spelling/codeql/blob/spell-check-codeql-python/.github/actions/spelling/line_forbidden.patterns)

Google has nearly 1/2 million hits for `decie`, it looks like it's a Polish (?) family/place name. I guess that Google hasn't decided that it's a typo for decide often enough to suggest it 😄.
There's definitely an argument to be made for increasing review proportional to the number of corrections in a line.

That said, if you'd like to skim through
https://github.com/check-spelling/codeql/blob/spell-check-codeql-python/.github/actions/spelling/expect/spell-check-codeql-python.txt
and see if anything jumps out at you...